### PR TITLE
INVGOV-13220 /like/{content_id} Node.js API endpoint

### PR DESCRIPTION
**Pull request by klauselhai**
https://io1dev.atlassian.net/browse/INVGOV-13220

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        in: path
+        required: true
+        schema:
+          type: string
+          format: uuid
+          example: '01234567-89ab-cdef-1234-56789abcdef0'
+    post:
+      tags:
+        - like
+      description: 'Mark a piece of content/comment as like.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5437]: 'Like a piece of content/comment.'

File: api.swagger.yaml [hunk 2]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+      responses:
+        '200':
+          description: 'The content/comment was like successfully.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5440]: 'Liked the content/comment successfully.'

File: api.swagger.yaml [hunk 3]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        '400':
+          description: 'The content/comment was not like due to malformed JSON.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5442]: like -> liked

File: api.swagger.yaml [hunk 4]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        '403':
+          description: 'User is not authorized to save a content/comment.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5444]: "save a" -> "like this"

File: api.swagger.yaml [hunk 5]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        '422':
+          description: 'The content/comment was not like due to Drupal validation failure or JSON schema validation failure.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5446]: like -> liked

File: api.swagger.yaml [hunk 6]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+      security:
+        - SessionCookieHeader: []
+        - APIKeyHeader: []
+        - CognitoJwt: [ invotra/like:write ]
+    delete:
+      tags:
+        - like
+      description: 'Unsave a piece of content/comment.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5454]: Unsave -> Unlike

File: api.swagger.yaml [hunk 7]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+          description: 'User is not authorized to save a content/comment.'
+        '422':
+          description: 'The content/comment was not like due to Drupal validation failure or JSON schema validation failure.'
+      security:
+        - SessionCookieHeader: []
+        - APIKeyHeader: []
+        - CognitoJwt: [ invotra/like:write ]
+    delete:
+      tags:
+        - like
```
Comment (ParryLuke) [inline api.swagger.yaml:5453]: "likes" category

File: api.swagger.yaml [hunk 8]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+      description: 'Unsave a piece of content/comment.'
+      responses:
+        '200':
+          description: 'The content/comment was unlike.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5457]: 'Like removed successfully.'

File: api.swagger.yaml [hunk 9]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        '400':
+          description: 'Could not unsave this content/comment due to malformed JSON.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5459]: unsave -> unlike

File: api.swagger.yaml [hunk 10]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        '403':
+          description: 'User is not authorized to unsave this content/comment.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5461]: unlike

File: api.swagger.yaml [hunk 11]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+        '422':
+          description: 'The content/comment was not unlike due to Drupal validation failure or JSON schema validation failure.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5463]: 'The like could not be removed due to...'

File: api.swagger.yaml [hunk 12]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+      security:
+        - SessionCookieHeader: []
+        - APIKeyHeader: []
+        - CognitoJwt: [ invotra/like:write ]
+    get:
+      description: 'Check if a piece of content/comment is like or not.'
```
Comment (ParryLuke) [inline api.swagger.yaml:5469]: like -> liked

File: api.swagger.yaml [hunk 13]
```diff
@@ -5421,6 +5421,82 @@ paths:
...
+      tags:
+        - like
```
Comment (ParryLuke) [inline api.swagger.yaml:5471]: likes category

File: api.swagger.yaml [hunk 14]
```diff
@@ -18802,3 +18878,9 @@ components:
         <html>
           ...
         </html>
+    content-like-get-response-schema:
+      type: object
+      properties:
+        current_user_like:
```
Comment (ParryLuke) [inline api.swagger.yaml:18884]: current_user_liked

**Review by ParryLuke — CHANGES_REQUESTED** [review]

**Review by ParryLuke — CHANGES_REQUESTED** [review]

File: api.swagger.yaml [hunk 15]
```diff
@@ -5421,6 +5421,82 @@ paths:
           description: The user does not have access to the requested content.
         '404':
           description: No content with that UUID was found.
+  '/like/{content_id}':
```
[the remark this answers is not in the document]
Comment (klauselhai) [reply api.swagger.yaml:5424]: @ParryLuke Sure, Alright.

**Review by ParryLuke — CHANGES_REQUESTED** [review]
Can we change the POST to a PUT?

I think we'd need to do the same for /saved but I'll check now

**Pull request merged**